File: cpp/src/hash_table.cpp
#include <utility>
#include <vector>
#include <unordered_map>

std::pair<int, int> twoSum(std::vector<int> nums, int target) {
    std::unordered_map<int, int> d;
    for (int i=0 ; i<nums.size() ; ++i) {
        if (d.contains(target - nums[i])) {
            return {d[target - nums[i]], i};
        } else {
            d[nums[i]] = i;
        }
    }
}

File: src/dlc/hash_table.py
from typing import List

def containsDuplicate(nums: List[int]) -> bool:
    """217. Contains Duplicate"""

    return len(nums) != len(set(nums))

def twoSum(nums, target):
    """1. Two Sum"""

    d = dict()
    for j, n in enumerate(nums):
        if target - n in d:
            return [d[target - n], j]
        else: 
            d[n] = j